src/common/helpers/estimated-tile-list.ts

```diff
@@ -87,11 +91,72 @@ const getTiles = (polygon: Polygon, minZoom: number, maxZoom: number): Array<str
             res.push([z, x, y].join('/') + '.png');
           }
         }
+
+        if (res.length % iterationsPerChunk === 0) {
+          await oneMoment();
+        }
       }
     }
     //   })
   });
   return res;
-}
+};
+
+const getTilesCount = (polygon: Polygon, minZoom: number, maxZoom: number): number => {
```

Comment:
consider reuse of those 2 functions parts instead of duplicating them

Reply:
That's the problem was, should be 2 of them, garbage collection just blows when generating tile images names.
Also for performance reason should be 2 of them